### ISSUE TITLE
Opening dataset without loading any indexes?

### ISSUE THREAD

**Issue by TomNicholas** [pydata/xarray#6633]
### What is your issue?

Within pangeo-forge's internals we would like to call `open_dataset`, then `to_dict()`, and end up with a schema-like representation of the contents of the dataset. This works, but it also has the side-effect of loading all indexes into memory, even if we are loading the data values "lazily". The end result would act a bit like `ncdump` does.

@benbovy do you think it would be possible to (perhaps optionally) also avoid loading indexes upon opening a dataset, so that we actually don't load anything?

cc @rabernat @jhamman @cisaacstern

**Comment by shoyer** [comment]
> > but another option (post explicit index refactor) might be an option for opening a dataset without creating indexes for 1D coordinates along dimensions.
> 
> It might indeed be worth considering this case too in #6392. Maybe `indexes=None` (default) to create default indexes for 1D coordinates and `indexes={}` (empty dictionary) to explicitly skip creating indexes?

+1 this syntax makes sense to me!

**Comment by rabernat** [comment]
Here is an example that really highlights the performance cost of always loading dimension coordinates:

```python
import zarr
store = zarr.storage.FSStore("s3://mur-sst/zarr/", anon=True)
%time list(zarr.open_consolidated(store)) # -> Wall time: 86.4 ms
%time ds = xr.open_dataset(store, engine='zarr') # -> Wall time: 17.1 s
```

`%prun` confirms that Xarray is spending most of its time just loading data for the `time` axis, which you can reproduce at the zarr level as:

```python
zgroup = zarr.open_consolidated(store)
%time _ = zgroup['time'][:] # -> Wall time: 14.7 s
```

Obviously this example is pretty extreme. But it really highlights the costs of eagerly loading dimension coordinates. If I don't care about label-based indexing for this dataset, I would rather have my 17s back!

:+1: to "`indexes={}` (empty dictionary) to explicitly skip creating indexes".

**Comment by shoyer** [comment]
Looking at this mur-sst dataset in particular, it stores time in chunks of size 5. That means fetching the 6443 time values requires 1288 separate HTTP requests -- no wonder it's so slow! If the time axis were instead stored in a single chunk of 51 KB, Xarray would only need 3 small size HTTP requests to load the lat, lon and time indexes, which would probably complete in a fraction of a second.

That said, I agree that this would be nice to have in general.

**Comment by rabernat** [comment]
Yes it is definitely a pathological example. 💣  But the fact remains that there are many cases where we just want to discover dataset contents as quickly as possible and want to avoid the cost of loading coordinates and creating indexes.

**Comment by dcherian** [comment]
This would also fix #2233

**Comment by dcherian** [comment]
Here's one from @lsetiawan that can't be opened because it has a 75GB `time` dimension coordinate.

```python
import zarr

zarr.open_group(
    "s3://ooi-data/RS03ECAL-MJ03E-06-BOTPTA302-streamed-botpt_nano_sample",
    mode="r+",
    storage_options=dict(anon=True),
)
```

**Comment by itcarroll** [comment]
Stumbled into this issue while experimenting with page_buf_size on the h5netcdf backend and looking for ways to get XArray closer to the speed of h5py for loading variables when coordinates are too much baggage. As an alternative to #8051, I would like to submit for consideration an "open_variable" method as a fast path from a store to an xarray.Variable (or a Mapping if given a list of variables in a store).
